packages/block-editor/src/components/inner-blocks/index.native.js

```diff
@@ -116,43 +108,31 @@ function UncontrolledInnerBlocks( props ) {
 
 	const BlockListComponent = useCompactList ? BlockListCompact : BlockList;
 
-	let blockList = (
-		<BlockListComponent
-			marginVertical={ marginVertical }
-			marginHorizontal={ marginHorizontal }
-			rootClientId={ clientId }
-			renderAppender={ renderAppender }
-			renderFooterAppender={ renderFooterAppender }
-			withFooter={ false }
-			orientation={ orientation }
-			parentWidth={ parentWidth }
-			horizontalAlignment={ horizontalAlignment }
-			horizontal={ horizontal }
-			contentResizeMode={ contentResizeMode }
-			contentStyle={ contentStyle }
-			onAddBlock={ onAddBlock }
-			onDeleteBlock={ onDeleteBlock }
-			filterInnerBlocks={ filterInnerBlocks }
-			gridProperties={ gridProperties }
-			blockWidth={ blockWidth }
-		/>
+	return (
+		<LayoutProvider value={ layout }>
+			<BlockContextProvider value={ context }>
+				<BlockListComponent
+					marginVertical={ marginVertical }
+					marginHorizontal={ marginHorizontal }
+					rootClientId={ clientId }
+					renderAppender={ renderAppender }
+					renderFooterAppender={ renderFooterAppender }
+					withFooter={ false }
+					orientation={ orientation }
+					parentWidth={ parentWidth }
+					horizontalAlignment={ horizontalAlignment }
+					horizontal={ horizontal }
+					contentResizeMode={ contentResizeMode }
+					contentStyle={ contentStyle }
+					onAddBlock={ onAddBlock }
+					onDeleteBlock={ onDeleteBlock }
+					filterInnerBlocks={ filterInnerBlocks }
+					gridProperties={ gridProperties }
+					blockWidth={ blockWidth }
+				/>
+			</BlockContextProvider>
+		</LayoutProvider>
 	);
-
-	// Wrap context provider if (and only if) block has context to provide.
-	const blockType = getBlockType( block.name );
-	if ( blockType && blockType.providesContext ) {
-		const context = getBlockContext( block.attributes, blockType );
-
-		blockList = (
-			<LayoutProvider value={ layout }>
```

Comment:
Seems to me you fixed a bug here: previously, if the component didn't provide a context, but did provide a layout, the `LayoutProvider` wasn't rendered and the layout wasn't applied.

I see quite a few blocks, very common ones, that are like this, i.e., their `block.json` has `supports.__experimentalLayout` field, but doesn't have `providesContext`. Buttons, Column, Columns, Group, ...

Reply:
Sort of. This was only working because `blockType && blockType.providesContext` was always truthy.

Reply:
Oh yes, so it was only potential, not real bug :+1: